In [3]:
import pandas as pd

sig_input = pd.read_csv("sig_input.csv")

sig_input

,variant,converted,not_converted,total
0,A,547,4446,4993
1,B,636,4371,5007


In [4]:
a = sig_input[sig_input.variant == "A"].iloc[0]
b = sig_input[sig_input.variant == "B"].iloc[0]
count = np.array([b.converted, a.converted])
nobs = np.array([b.total, a.total])
p_b, p_a = count[0]/nobs[0], count[1]/nobs[1]
p_pool = count.sum() / nobs.sum()
se = np.sqrt(p_pool * (1- p_pool) * (1/nobs[0] + 1/nobs[1]))
z = (p_b- p_a) / se
p_val = 2 * (1- stats.norm.cdf(abs(z)))
rel_lift = (p_b- p_a) / p_a * 100

In [5]:
p_val

np.float64(0.00684165923124791)

In [6]:
rel_lift

np.float64(15.94546428418861)

In [7]:
import pandas as pd

df = pd.read_csv("one8_ab_test_sessions.csv")

In [8]:
df[["variant", "revenue"]].head()

,variant,revenue
0,A,0.0
1,A,0.0
2,A,0.0
3,A,0.0
4,A,0.0


In [9]:
from scipy import stats

rev_a = df[df.variant == "A"].revenue
rev_b = df[df.variant == "B"].revenue

t_stat, p_rev = stats.ttest_ind(
    rev_b,
    rev_a,
    equal_var=False
)

In [10]:
print("===== REVENUE A/B TEST =====")
print(f"A mean revenue: ₹{rev_a.mean():.2f}")
print(f"B mean revenue: ₹{rev_b.mean():.2f}")
print(f"Revenue difference: ₹{rev_b.mean() - rev_a.mean():.2f}")
print(f"t-statistic: {t_stat:.3f}")
print(f"p-value: {p_rev:.4f}")

===== REVENUE A/B TEST =====
A mean revenue: ₹377.42
B mean revenue: ₹425.08
Revenue difference: ₹47.66
t-statistic: 1.632
p-value: 0.1027


Variant B has a higher observed conversion rate than A. The two-proportion z-test produces a p-value of 0.00684, indicating that the difference is statistically significant at the 5% level.
